c7n/resources/support.py

```diff
@@ -46,3 +47,51 @@ class resource_type(TypeInfo):
     source_mapping = {
         "describe": DescribeAdvisorCheck,
     }
+
+
+@AdvisorCheck.filter_registry.register('resource-status')
+class StatusErrorFilter(Filter):
+    """Filter Trusted Advisor flagged resources that match specific statuses
```

Comment:
why does this need a separate filter? it looks like it would be covered by the generic `list-item` filter? https://cloudcustodian.io/docs/filters.html#list-item-filter

Reply:
@kapilt Initially I tried to work with the list-item filter, but the flaggedResources list returned, still included all resources evaluated.  It successfully filtered on the list of Trusted Advisor recommendations returned, based on the status of the flaggedResources specified in the list-item filter.  However, I couldn't find a way to discard flaggedResources that did not match the status.  An example policy I used is as follows:

`policies:
- name: trusted-advisor
  resource: advisor-check
  filters:
  - type: value
    key: status
    op: in
    value: "error"
  - type: list-item
    key: flaggedResources[]
    attrs:
    - type: value
      key: status
      op: in
      value: "error"`
      
In the policy above, I would like to return only recommendations that have a status of error and return flaggedResources associated with the recommendation that are in error too.  However, the flaggedResources returned included those with a status of warning and ok as well.  Considering the combination of value filter and list-item filter didn't accomplish what I was looking to do, I added this filter.